base62.go

```diff
@@ -1,17 +1,17 @@
 package ksuid
 
-import "bytes"
+import "strings"
 
 // lexographic ordering (based on Unicode table) is 0-9A-Za-z
-var base62Characters = []byte("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")
+var base62Characters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
 
-func base2base(in []byte, inBase int, outBase int) []byte {
-	res := []byte{}
-	bs := in[:]
+func appendBase2Base(dst []byte, src []byte, inBase int, outBase int) []byte {
+	bs := src[:]
+	bq := [stringEncodedLength]byte{}
```

Comment:
So the original base2base function is generic: it works with arbitrary-length bytestrings. This change appears to maybe violate that in a way that is pretty sneaky? From what I'm reading it looks like `bq` is here to avoid allocating a `[]byte` for each run of the `for` loop below and that it's pre-set to the size of the stringEncodedLength to avoid allocations from resizing the slice during appends. If a longer string comes in here is this just harmless (it's given the capacity of the size used in this library) or will it just panic and die?